### PR TITLE
Refactoring of ZCL_ABAPGIT_ENVIRONMENT

### PR DESCRIPTION
**Pull request by ThomasPloski**
Refactoring is needed in order to be able to mock this central abapGIT environment class in unit tests, so that different system states can be tested within unit tests.

### CODE REVIEW

File: src/zcl_abapgit_environment.clas.abap [hunk 1]
```diff
@@ -1,82 +1,83 @@
...
       ENDIF.
     ENDIF.
-    rv_allowed = gv_client_modifiable.
+    rv_result = mv_client_modifiable.
   ENDMETHOD.
 
 
-  METHOD is_sap_cloud_platform.
+  METHOD zif_abapgit_environment~is_restart_required.
+    rv_result = abap_false.
```
Comment (larshp) [inline src/zcl_abapgit_environment.clas.abap]: please add comment describing where this is used, currently it risks being deleted

File: src/zcl_abapgit_objects.clas.abap [hunk 2]
```diff
@@ -1113,8 +1114,12 @@ CLASS ZCL_ABAPGIT_OBJECTS IMPLEMENTATION.
     li_obj = create_object( is_item     = rs_files_and_item-item
                             iv_language = iv_language ).
     li_obj->mo_files = lo_files.
+
     CREATE OBJECT lo_xml.
-    li_obj->serialize( lo_xml ).
+    CREATE OBJECT lo_log.
+
+    li_obj->serialize( io_xml      = lo_xml
+                       ii_log      = lo_log ). "to be replaced by item logger
```
Comment (larshp) [inline src/zcl_abapgit_objects.clas.abap]: this change is unrelated, please remove

and will cause syntax errors in master

File: src/zcl_abapgit_environment.clas.abap (hunk 1 above)
Comment (ThomasPloski) [reply src/zcl_abapgit_environment.clas.abap]: comment is added

File: src/zcl_abapgit_objects.clas.abap (hunk 2 above)
Comment (ThomasPloski) [reply src/zcl_abapgit_objects.clas.abap]: signature changes so that no log is required to pass
